js/src/ui/Form/AddressSelect/addressSelectStore.js

```diff
@@ -0,0 +1,180 @@
+// Copyright 2015, 2016 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+import { observable, action } from 'mobx';
+import { flatMap } from 'lodash';
+
+import Contracts from '~/contracts';
+import { sha3 } from '~/api/util/sha3';
+
+export default class AddressSelectStore {
+
+  @observable values = [];
+  @observable regsitryValues = [];
+
+  initValues = [];
+  regLookups = [];
+
+  constructor (api) {
+    this.api = api;
+
+    const { registry } = Contracts.get();
+
+    registry
+      .getContract('emailverification')
+      .then((emailVerification) => {
+        this.regLookups.push({
+          lookup: (value) => {
+            return emailVerification
+              .instance
+              .reverse.call({}, [ sha3(value) ]);
+          },
+          describe: (value) => `${value} (from email verification)`
```

Comment:
I'd prefer "Verified using email …", as this is being shown inside and account card.